src/main/java/frc/robot/subsystems/vision/SImVisionIO.java

```diff
@@ -35,29 +38,35 @@ private SImVisionIO(int camIndex) {
         }
     }
 
-
     @Override
     public void setPipeLine(int pipeLineIndex) {
         simCamera.getCamera().setPipelineIndex(pipeLineIndex);
     }
 
     @Override
     public Result getLatestResult() {
-        return new Result(simCamera.getCamera().getLatestResult().getTimestampSeconds(), new Pose3d(simCamera.getCamera().getLatestResult().getBestTarget().getBestCameraToTarget().getTranslation(),simCamera.getCamera().getLatestResult().getBestTarget().getBestCameraToTarget().getRotation()));
+        return new Result(simCamera.getCamera().getLatestResult().getTimestampSeconds(), new Pose3d(simCamera.getCamera().getLatestResult().getBestTarget().getBestCameraToTarget().getTranslation(), simCamera.getCamera().getLatestResult().getBestTarget().getBestCameraToTarget().getRotation()));
```

Comment:
`simCamera.getCamera().getLatestResult()` should be extracted into a variable, for two reasons:
- Reduce code repittion.
- It's possible that between the first and the second/third call of `getLatestResult()` another result will come in. This is problematic because the returned `Result` should have data from exactly one measurement -- we don't want the `Result` to have a coordinate of `t1` but timestamp of `t2`.

But I must ask, why is this function even needed? Why not just use `getLatestResult()` directly? All your function does is remove information from the value returned from that function.